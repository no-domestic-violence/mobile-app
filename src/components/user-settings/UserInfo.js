import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import UserSettingsSVG from '_assets/svg/userSettings.svg';
import { faUserCircle } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';

export default function UserInfo({ username }) {
  return (
    <View>
      <UserSettingsSVG style={{ position: 'absolute' }} />
      <View style={styles.userInfo}>
        <FontAwesomeIcon
          icon={faUserCircle}
          size={80}
          color={'grey'}
          style={styles.arrow}
        />
        <Text style={styles.username}>{username}</Text>
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  username: {
    textTransform: 'capitalize',
    fontSize: 28,
    marginLeft: 15,
  },
  userInfo: {
    flexDirection: 'row',
    alignItems: 'center',
    marginLeft: 10,
    marginTop: 10,
  },
});
