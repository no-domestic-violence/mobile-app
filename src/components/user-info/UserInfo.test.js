import React from 'react';
import { render } from '@testing-library/react-native';
import UserInfo from './index';

describe('User information component', () => {
  const username = 'Celeste';
  it('renders name of user from props', () => {
    const { queryByText } = render(<UserInfo username={username} />);
    expect(queryByText(username)).not.toBeNull();
  });
  it('should match snapshot', () => {
    const result = render(<UserInfo username={username} />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
