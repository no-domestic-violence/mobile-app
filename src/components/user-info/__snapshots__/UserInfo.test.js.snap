// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User information component should match snapshot 1`] = `
<View>
  <svg
    data-file-name="SvgUserSettings"
    style={
      Object {
        "position": "absolute",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 10,
        "marginTop": 10,
      }
    }
  >
    <
      color="grey"
      icon={
        Object {
          "icon": Array [
            496,
            512,
            Array [],
            "f2bd",
            "M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z",
          ],
          "iconName": "user-circle",
          "prefix": "fas",
        }
      }
      size={80}
    />
    <Text
      style={
        Object {
          "fontSize": 28,
          "marginLeft": 15,
          "textTransform": "capitalize",
        }
      }
    >
      Celeste
    </Text>
  </View>
</View>
`;
